bundles/org.openhab.binding.satel/src/main/java/org/openhab/binding/satel/internal/protocol/SatelModule.java

```diff
@@ -52,11 +54,12 @@
 
     private final BlockingQueue<SatelCommand> sendQueue = new LinkedBlockingQueue<SatelCommand>();
 
-    private IntegraType integraType;
-    private int timeout;
-    private String integraVersion;
+    private final int timeout;
+    private volatile IntegraType integraType;
+    private volatile String integraVersion;
+    private volatile boolean extPayloadSupport;
     private CommunicationChannel channel;
-    private Object channelLock;
+    private final Object channelLock = new Object();
```

Comment:
I think you need the lock to protect the channel from disconnecting two times. Wouldn't be a `Lock` be better? If `tryLock()` fails the thread would not be blocked. Or is blocking until disconnect what you want to achieve?

Reply:
Good question. :-)
This is very old code and I don't remember what was the purpose. It seems to be useless as the whole `disconnect` method is synchronized. I think I was about to remove synchronization from the method and use this `channelLock` to not lock the whole object, but just `disconnect` method. 
But now I am not sure whether or not it is safe to remove this lock from the code and I would't do this in this PR.